hypothesis/lms
# Missing optional parameters in LTI 1.3 parameter mapping

From this slack thread, something was wrong with the email field in LTI1.3.


It turns out that the field was entirely missing. I reckon I missed this as:

- email is optional
- it's only stored on `group_info` which  is a bit on the "sidelines" of the LMS app.


Took the opportunity to also find some missing tool_platform 1.3 parameters.


# Testing

- Clean the group_info table

tox -qe dockercompose -- exec postgres psql -U postgres -c "truncate group_info cascade"


- Launch a localhost lti1.3 assignment 

https://hypothesis.instructure.com/courses/319/assignments/3308



- Email is stored in `group_info`

```
tox -qe dockercompose -- exec postgres psql -U postgres -c "select info->'instructors' from group_info"
````


### Tool parameters


Canvas only sends the `version` tool_platform parameter, from the JWT in the same assigmetn:

```
"https://purl.imsglobal.org/spec/lti/claim/tool_platform": {
    "guid": "VCSy8DAKCFIQYaItpSGyXhOXIlk5A6NWduVVG1u3:canvas-lms",
    "name": "University of Hypothesis",
    "version": "cloud",
    "product_family_code": "canvas",
    "validation_context": null,
    "errors": {
      "errors": {}
    }
  },
```

- Remove that value:

```
tox -qe dockercompose -- exec postgres psql -U postgres -c "update application_instances set tool_consumer_info_version = null where id = 102"
```


- Launch again https://hypothesis.instructure.com/courses/319/assignments/3308

- Check that the value is stored:

```
tox -qe dockercompose -- exec postgres psql -U postgres -c "select tool_consumer_info_version from application_instances where id = 102"

 tool_consumer_info_version 
----------------------------
 cloud
(1 row)
```
